gigascience/gigadb-website
# Fix color contrast issues in pages with new layouts

# Pull request for issue: #1363 

This is a pull request for the following functionalities:

* Fix most color contrast issues in all pages that use the newer layouts (hence, the newer styles) (most, not all, because a few have been fixed in separate PRs)

## How to test?

You can run axe-cli and grep wht results for `color-contrast`, here's how to do it:

- Install axe-cli: https://github.com/dequelabs/axe-core-npm/blob/develop/packages/cli/README.md
- Run gigadb project locally
- Run `axe http://gigadb.gigasciencejournal.com:9170` to check that it works
- If you get some error related to chromedriver:
  - Download latest chromedriver (or the one that matches your chrome version) from here https://googlechromelabs.github.io/chrome-for-testing/ or here https://chromedriver.chromium.org/downloads
  - Run `axe http://gigadb.gigasciencejournal.com:9170 --chromedriver-path="/your-path-to-local-chromedriver/chromedriver"`
- Run something like the following in the console:

```sh
axe http://gigadb.gigasciencejournal.com:9170/dataset/100006 >> temp-axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/ >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/contact >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/help >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/guide >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/guidegenomic >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/faq >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/about >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/team >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/advisory >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/term >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/login >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/forgot >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/site/thanks >> axe-cli-audit.txt
axe http://gigadb.gigasciencejournal.com:9170/user/create >> axe-cli-audit.txt
```

These are all distinct public pages that use the new layouts, except for the dataset listing page whose contrast issues are fixed in a different PR

- Grep the log for `color-contrast` and verify no issues are found

- Inspect this page manually using Axe chrome extension or similar, since it requires login: http://gigadb.gigasciencejournal.com:9170/user/view_profile

## How have functionalities been implemented?

* The colors in file less/current/base/variables.less have been revamped:
  * Gigadb green has been updated to fulfill color contrast criterion (text on background contrast)
  * Previously existing color variables left untouched
  * Added new color variations for green and red, left unused ones commented out in case they need to be used in the future (the plan is to eventually get rid of the unused colors)
  * Defined a serie of variables that fulfill color contrast criteria in an obvious way so that this condition is more explicit when using styles
* Updated all styles present in the pages that use the new layouts, to handle color contrast issues
* Use the current.css file in the new_faq layout to leverage the updated colors
* Removed hardcoded hex color values from less/current/**/*.less files, replacing them by color variables

Note: the style guide should be updated to reflect these changes, probably better left to a different ticket

## Any issues with implementation?

None

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
